data/data/bootstrap/baremetal/files/usr/local/bin/master-bmh-update.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+. /usr/local/bin/release-image.sh
+
+wait_for_existence() {
+    while [ ! -e "${1}" ]
+    do
+        sleep 5
+    done
+}
+
+# Wait for the API to become active
+until curl -s http://localhost:6385/v1/nodes; do
+    echo Waiting for API
+    sleep 20
+done
+
+# Wait for the nodes to become active after introspection.
+while curl -s http://localhost:6385/v1/nodes | jq '.nodes[] | .provision_state' | grep -v active; do
+    echo Waiting for nodes to become active
+    sleep 20
+done
```

Comment:
There's a race here, we could query for nodes before there are any:

```
Waiting for API
Waiting for API
Waiting for API
Waiting for API
Waiting for API
{"nodes": []}Nodes are all active
```

I think maybe the previous check should be to see if there's any nodes

Reply:
Yeah I've seen this too. I'm testing with your above check only.. I think that would be better.

Reply:
I think there's still a race, the BareMetalHosts exist doesn't mean there's hosts in Ironic.  The timing is probably such that it works, but an explicit check that Ironic has at least 3 nodes would be safer.

Reply:
yeah just saw this again.. will add another check